Feat/prof page

heartgg opened the pull request:
Related to #6 
More description soon

LudoLogical commented on src/components/Card.tsx:
```diff
@@ -0,0 +1,7 @@
+export const Card = (props) => {
```
```suggestion
export const Card = ({children}) => {
```
Use destructuring whenever not unreasonable!

LudoLogical replied:
Also, instead of `export const Card`, consider `const Card` and then use `export default Card;` to leverage default exports! This applies throughout the PR

LudoLogical commented on src/components/HorizontalScores.tsx:
```diff
@@ -0,0 +1,24 @@
+interface Scores {
```
```suggestion
type Scores = {
```
Prefer types over interfaces when reasonable, they are more extensible

LudoLogical commented on src/components/HorizontalScores.tsx:
```diff
@@ -0,0 +1,24 @@
+  wtaPercent: number;
+}
+
+export const HorizontalScores = ({ rmpScore, diffScore, wtaPercent }:Scores) => {
```
```suggestion
export const HorizontalScores = ({
  rmpScore,
  diffScore,
  wtaPercent
}: Scores) => {
```
Code cleanliness!

LudoLogical commented on src/components/HorizontalScores.tsx:
```diff
@@ -0,0 +1,24 @@
+  const getScoreColor = (value: number, outOf: number) => {
+    value /= outOf;
+    const hue = (value * 120).toString(10);
+    return ["hsl(", hue, ",100%,85%)"].join("");
```
Add a commend explaining what this does?

LudoLogical commented on src/components/HorizontalScores.tsx:
```diff
@@ -0,0 +1,24 @@
+
+  return (
+    <div className="grid grid-cols-3">
+      <h2 className="bg-blue-dark text-white text-center py-1 rounded-tl-2xl">RMP</h2>
```
Pull the repeated tailwindcss out of here and put it in a const variable. Then, use concatenation to add the extra in! It'll be much neater

LudoLogical commented on src/components/ProfileGrades.tsx:
```diff
@@ -0,0 +1,97 @@
+import Chart from "react-apexcharts"
```
It's good practice to add ; here!

LudoLogical commented on src/components/ProfileGrades.tsx:
```diff
@@ -0,0 +1,97 @@
+import Chart from "react-apexcharts"
+import { useState } from "react"
+
+export interface GradeDistribution {
```
Again, prefer types if you can!

LudoLogical commented on src/components/ProfileGrades.tsx:
```diff
@@ -0,0 +1,97 @@
+  series: ApexAxisChartSeries;
+}
+
+import { AiFillCaretLeft, AiFillCaretRight } from "react-icons/ai"
```
Group all of your imports at the top!

LudoLogical commented on src/components/ProfileGrades.tsx:
```diff
@@ -0,0 +1,97 @@
+
+import { AiFillCaretLeft, AiFillCaretRight } from "react-icons/ai"
+
+const gradeChartOptions = {
```
There sure is a lot here! Good formatting putting things on new lines, but maybe this stuff belongs in a separate config TypeScript file that you can then import this const var from?

LudoLogical commented on src/components/ProfileGrades.tsx:
```diff
@@ -0,0 +1,97 @@
+export const ProfileGrades = (props) => {
+  const [page, setPage] = useState(0);
+
+  const { gradeDistributionData }:{ gradeDistributionData: GradeDistribution[] } = props;
```
```suggestion
  const { gradeDistributionData }: { gradeDistributionData: GradeDistribution[] } = props;
```
Adding a space after your colons dramatically improves legibility!

LudoLogical commented on src/components/ProfileGrades.tsx:
```diff
@@ -0,0 +1,97 @@
+
+  return (
+    <>
+      <header className="bg-blue-dark rounded-t-2xl flex">
```
Again, these classNames needn't repeat so much if you extract the common stuff in variable(s)!

LudoLogical commented on src/routes/about.tsx:
```diff
@@ -1,10 +1,48 @@
 export const About = () => {
   const navigation: NavigateFunction = useNavigate()
 
+  const mockData: ProfessorProfileInterface = {
```
Always separate your mock data from the place where you are using it! This helps make sure you are passing variables to components the right way.

LudoLogical reviewed (commented):
Hey, React expert here. I'm not qualified to approve or reject these changes from a functionality perspective, but David (Head of Engineering) asked me to make some style and good practice suggestions. In general, the comments I've left apply throughout all code changes. Hope this helps, and thanks for reaching out!

heartgg commented:
@LudoLogical Thank you very much for spending the time to review the PR. I greatly appreciate it! I addressed most of the issues, but I left some that I thought would be solved in the future via code prettiness plugins, linting, etc. I took note of them and will start working on getting those plugins into the codebase and required before merging PRs (via CI/CD). I also left Tailwind styling as is (even though sometimes it's repetitive) per the Tailwind documentation.

LudoLogical commented:
@heartgg You're very welcome; thank you for considering the review! It's great that you're working on setting up things like eslint and prettier. Leaving the Tailwind as-is is perfectly reasonable. In fact, instead of putting things in variables, the official line is actually that, for large projects, anything with repeated Tailwind styles should be considered for componentization (e.g., <Heading1> instead of <h1 className="...">). Just wanted to throw that out there in case it ends up being right for you down the line.

dlaunikitis reviewed (approved):
LGTM
